tests/components/plaato/test_config_flow.py

```diff
@@ -0,0 +1,234 @@
+    ):
+        yield
+
+
+async def init_config_flow(hass):
+    """Init a configuration flow."""
+    await async_process_ha_core_config(
+        hass, {"external_url": BASE_URL},
+    )
+    flow = config_flow.PlaatoConfigFlow()
+    flow.hass = hass
+    return flow
+
+
+async def test_show_config_form(hass):
+    """Test show configuration form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "device_type"
+
+
+async def test_show_config_form_device_type_airlock(hass):
+    """Test show configuration form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": "device_type"},
+        data={
+            CONF_DEVICE_TYPE: PlaatoDeviceType.Airlock,
+            CONF_DEVICE_NAME: "device_name",
+        },
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "validate"
+    assert result["data_schema"].schema.get(CONF_TOKEN) == str
+    assert result["data_schema"].schema.get(CONF_USE_WEBHOOK) == bool
+
+
+async def test_show_config_form_device_type_keg(hass):
+    """Test show configuration form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": "device_type"},
+        data={CONF_DEVICE_TYPE: PlaatoDeviceType.Keg, CONF_DEVICE_NAME: "device_name"},
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "validate"
+    assert result["data_schema"].schema.get(CONF_TOKEN) == str
+    assert result["data_schema"].schema.get(CONF_USE_WEBHOOK) is None
+
+
+async def test_show_config_form_validate_webhook(hass, webhook_id):
+    """Test show configuration form."""
+
+    flow = await init_config_flow(hass)
+    flow._init_info = {CONF_DEVICE_TYPE: PlaatoDeviceType.Airlock}
+
+    result = await flow.async_step_validate(
+        user_input={CONF_USE_WEBHOOK: True, CONF_TOKEN: ""}
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "webhook"
+
+    flow._init_info = {CONF_DEVICE_TYPE: PlaatoDeviceType.Keg}
+
+    result = await flow.async_step_validate(
+        user_input={CONF_USE_WEBHOOK: True, CONF_TOKEN: ""}
+    )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "validate"
+    assert len(flow._errors) == 1
+    assert flow._errors["base"] == "invalid_webhook_device"
+
+
+async def test_show_config_form_validate_token(hass):
+    """Test show configuration form."""
+
+    flow = await init_config_flow(hass)
+    flow.context = {}
+    flow._init_info = {CONF_DEVICE_TYPE: PlaatoDeviceType.Airlock}
+
+    result = await flow.async_step_validate(
+        user_input={CONF_USE_WEBHOOK: False, CONF_TOKEN: "token"}
+    )
+
```

Comment:
Please mock out `async_setup` and `async_create_entry` here so the entry isn't actually setup